client/state/notices/account-recovery/index.js

```diff
@@ -6,9 +6,43 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import { errorNotice } from 'state/notices/actions';
+import {
+	successNotice,
+	errorNotice,
+} from 'state/notices/actions';
+
+import {
+	getAccountRecoveryPhone,
+	getAccountRecoveryEmail,
+} from 'state/account-recovery/settings/selectors';
+
 import { dispatchError } from '../utils';
 
+const getUpdateSuccessMessage = ( target, getState ) => {
+	switch ( target ) {
+		case 'phone':
+			const oldPhone = getAccountRecoveryPhone( getState() );
+
+			if ( null == oldPhone ) {
+				return translate( 'Successfully added new recovery SMS number.' );
+			}
+
+			return translate( 'Successfully updated recovery SMS number.' );
+
+		case 'email':
+			const oldEmail = getAccountRecoveryEmail( getState() );
+
+			if ( ! oldEmail ) {
+				return translate( 'Successfully added new recovery email address.' );
+			}
+
+			return translate( 'Successfully updated recovery email address.' );
+
+		default:
+			return translate( 'Successfully updated the recovery option.' );
```

Comment:
Sorry for the confusion. I've opened a PR for adding the explanation: #10396 . Could you help me verify if that clarifies its purpose clear enough?